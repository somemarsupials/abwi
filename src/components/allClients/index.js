export { default as ClientsPage } from './clientsList';
export { default as ClientsList } from './clientsPage';
