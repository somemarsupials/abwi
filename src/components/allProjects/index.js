export { default as List } from './list';
export { default as Page } from './page';
