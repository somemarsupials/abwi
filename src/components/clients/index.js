export { default as ClientPage } from './clientPage';
export { default as ClientDetail } from './clientDetail';
