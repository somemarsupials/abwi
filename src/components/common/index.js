export { default as TitleBar } from './titleBar';
export { default as ErrorBar } from './errorBar';
