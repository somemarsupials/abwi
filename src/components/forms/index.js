export { default as ProjectForm } from './projectForm';
