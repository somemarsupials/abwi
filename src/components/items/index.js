export { default as ItemList } from './itemList';
