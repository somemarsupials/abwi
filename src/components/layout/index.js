export { default as SubHeader } from './subHeader';
