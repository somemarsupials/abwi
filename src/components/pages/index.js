export { default as ClientsPage } from './clientsPage';
export { default as ClientPage } from './clientPage';
export { default as ProjectsPage } from './projectsPage';
export { default as ProjectPage } from './projectPage';
