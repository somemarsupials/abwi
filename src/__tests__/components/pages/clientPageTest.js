import React from 'react';
import { shallow } from 'enzyme';
import { ClientPage } from '../../../components/pages';

describe('ClientPage', () => {
  let page;
  let client;

  beforeEach(() => {
    client = 'client';
  });

  describe('with error', () => {
    beforeEach(() => {
      page = shallow(<ClientPage error={'an error'}/>);
    });

    it('renders with error bar', () => {
      expect(page).toMatchSnapshot();
    });
  });

  describe('when loading', () => {
    beforeEach(() => {
      page = shallow(<ClientPage fetching={true} />);
    });

    it('renders with loading information', () => {
      expect(page).toMatchSnapshot();
    });
  });

  describe('when given clients', () => {
    beforeEach(() => {
      page = shallow(<ClientPage client={client} />);
    });

    it('correctly renders', () => {
      expect(page).toMatchSnapshot();
    });
  });
});

