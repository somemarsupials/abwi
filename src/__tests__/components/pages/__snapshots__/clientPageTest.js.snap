// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientPage when given clients correctly renders 1`] = `
<div>
  <PageHeader
    bsClass="page-header"
  />
  <div>
    <Component>
      Projects
    </Component>
    <Component />
  </div>
</div>
`;

exports[`ClientPage when loading renders with loading information 1`] = `
<div>
  <PageHeader
    bsClass="page-header"
  />
  <Alert
    bsClass="info"
    bsStyle="info"
    closeLabel="Close alert"
  >
    Fetching client...
  </Alert>
</div>
`;

exports[`ClientPage with error renders with error bar 1`] = `
<div>
  <PageHeader
    bsClass="page-header"
  />
  <Alert
    bsClass="alert"
    bsStyle="info"
    closeLabel="Close alert"
  >
    Could not load client | 
    an error
  </Alert>
</div>
`;
